src/runtime/internal/database.server.ts

```diff
@@ -61,12 +61,30 @@ export async function checkAndImportDatabaseIntegrity(event: H3Event, collection
 async function _checkAndImportDatabaseIntegrity(event: H3Event, collection: string, integrityVersion: string, config: RuntimeConfig['content']) {
   const db = loadDatabaseAdapter(config)
 
-  const before = await db.first<{ version: string }>(`select * from ${tables.info} where id = ?`, [`checksum_${collection}`]).catch(() => ({ version: '' }))
+  const before = await db.first<{ version: string, ready: boolean }>(`select * from ${tables.info} where id = ?`, [`checksum_${collection}`]).catch(() => ({ version: '', ready: true }))
```

Comment:
At first glance it is hard to understand why `read` is `true` in catch.
it would be better if we return `null` or `undefined` in catch and move the interval logic inside `if (before?.version) {`.

```ts

const before = await db.first<{ version: string, ready: boolean }>(`select * from ${tables.info} where id = ?`, [`checksum_${collection}`]).catch(() => null)

if (before?.version) {
  if (before.ready === true && before.version === integrityVersion) {
    return true
  }
  
  if (before.ready === false && before.version === integrityVersion) {
    // Loop 
  }
  
  // Delete old version -- checksum exists but does not match with bundled checksum
  await db.exec(`DELETE FROM ${tables.info} WHERE id = ?`, [`checksum_${collection}`])
}
```